.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,234 @@
+# This GitHub workflow config has been generated by a script via
+#
+#   haskell-ci 'github' '--config=cabal.haskell-ci' 'crypto-rng-effectful.cabal'
+#
+# To regenerate the script (for example after adjusting tested-with) run
+#
+#   haskell-ci regenerate
+#
+# For more information, see https://github.com/haskell-CI/haskell-ci
+#
+# version: 0.15.20220620
+#
+# REGENDATA ("0.15.20220620",["github","--config=cabal.haskell-ci","crypto-rng-effectful.cabal"])
+#
+name: Haskell-CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  linux:
+    name: Haskell-CI - Linux - ${{ matrix.compiler }}
+    runs-on: ubuntu-20.04
+    timeout-minutes:
+      60
+    container:
+      image: buildpack-deps:bionic
+    continue-on-error: ${{ matrix.allow-failure }}
+    strategy:
+      matrix:
+        include:
+          - compiler: ghc-9.2.3
+            compilerKind: ghc
+            compilerVersion: 9.2.3
+            setup-method: ghcup
+            allow-failure: false
+          - compiler: ghc-9.0.2
+            compilerKind: ghc
+            compilerVersion: 9.0.2
+            setup-method: ghcup
+            allow-failure: false
+          - compiler: ghc-8.10.7
+            compilerKind: ghc
+            compilerVersion: 8.10.7
+            setup-method: ghcup
+            allow-failure: false
+          - compiler: ghc-8.8.4
+            compilerKind: ghc
+            compilerVersion: 8.8.4
+            setup-method: hvr-ppa
```

Comment:
Any reason why we're still using hvr-ppa instead of going full on ghcup?

Reply:
`haskell-ci` autogenerates these. You can force usage of ghcup in `cabal.haskell-ci`, but as long as it works, it doesn't matter to me ;)